client/v3/concurrency/session.go

```diff
@@ -116,6 +120,8 @@ func WithTTL(ttl int) SessionOption {
 	return func(so *sessionOptions) {
 		if ttl > 0 {
 			so.ttl = ttl
+		} else {
+			logger.Info("TTL should be > 0, defaulting to 60")
```

Comment:
Please move this check to `NewSesson` function after options are applied.

Reply:
@serathius alternative is to add logger as SessionOption param
`type SessionOption func(*sessionOptions, *zap.Logger)`
and then `opt(ops, client.GetLogger())`

Reply:
This sounds more intuitive lemme give this a try.